File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice= prices[0], maxmProfit=0;
        for(int i=1;i<(int)prices.size();++i){
            maxmProfit= max(maxmProfit, prices[i]-minPrice);
            minPrice= min(minPrice, prices[i]);
        }
        return maxmProfit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n= prices.length;
        if(n==0) return 0;
        int currMin= prices[0], maximumProfit=0;
        for(int i=1;i<n;i++){
            maximumProfit= Math.max(prices[i]-currMin, maximumProfit);
            currMin= Math.min(currMin, prices[i]);
        }
        return maximumProfit;
    }
}